src/test/java/org/sonarlint/intellij/analysis/AnalysisStateTests.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015-2024 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.analysis
+
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.vfs.VirtualFile
+import java.util.UUID
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito
+import org.sonarlint.intellij.AbstractSonarLintLightTests
+import org.sonarlint.intellij.trigger.TriggerType
+
+class AnalysisStateTests : AbstractSonarLintLightTests() {
+
+    private val file1 = Mockito.mock(VirtualFile::class.java)
+    private val file2 = Mockito.mock(VirtualFile::class.java)
+    private val callback = Mockito.mock(AnalysisCallback::class.java)
+    private val progress = Mockito.mock(ProgressIndicator::class.java)
+    private lateinit var randomUuid1: UUID
+    private lateinit var randomUuid2: UUID
+
+    @BeforeEach
+    fun init() {
+        randomUuid1 = UUID.randomUUID()
+        randomUuid2 = UUID.randomUUID()
+    }
+
+    @Test
+    fun should_find_redundant_analysis_for_non_snapshot() {
```

Comment:
I think we could add a medium test that starts an actual analysis to make sure we are canceling the old analysis

Reply:
I tried but I had difficulties making a test that triggered a long enough analysis that I could cancel it with a new one